sotodlib/preprocess/processes.py

```diff
@@ -1490,6 +1487,162 @@ def process(self, aman, proc_aman):
+            self.save(proc_aman, rot_aman)
+        
+    def calc_and_save(self, aman, proc_aman):
+        if self.calc_cfgs is None:
+            self.calc_cfgs = {}
+        if self.calc_cfgs:
+            deproj_aman = core.AxisManager(aman.dets, aman.samps)
+            deproj_demodQ, deproj_demodU = tod_ops.deproject.medQU_correct(aman) #, **self.calc_cfgs)
+            deproj_aman.wrap("deproj_demodQ", deproj_demodQ, [(0, 'dets'), (1, 'samps')])
+            deproj_aman.wrap("deproj_demodU", deproj_demodU, [(0, 'dets'), (1, 'samps')])
+            self.save(proc_aman, deproj_aman)
+        
+    def save(self, proc_aman, deproj_aman):
+        if self.save_cfgs is None:
+            return
+        if self.save_cfgs:
+            proc_aman.wrap("deproj", deproj_aman)
+
+class FocalplaneFlags(_Preprocess):
+    """Find additional detectors which have nans 
+       in their focal plane coordinates.
+
+    Saves results in proc_aman under the "fp_flags" field. 
+
+     Example config block::
+
+        - name : "fp_flags"
+          signal: "signal" # optional
+          calc: True
+          save: True
+          select: True
+    
+    .. autofunction:: sotodlib.tod_ops.flags.get_focalplane_flags
+    """
+    name = "fp_flags"
+    
+    def calc_and_save(self, aman, proc_aman):
+        mskfp = tod_ops.flags.get_focalplane_flags(aman, merge=False)
+        
+        fp_aman = core.AxisManager(aman.dets, aman.samps)
+        fp_aman.wrap('darks', mskfp, [(0, 'dets'), (1, 'samps')])
+        self.save(proc_aman, fp_aman)
+    
+    def save(self, proc_aman, fp_aman):
+        if self.save_cfgs is None:
+            return
+        if self.save_cfgs:
+            proc_aman.wrap("fp_flags", dark_aman)
+    
+    def select(self, meta, proc_aman=None):
+        if self.select_cfgs is None:
+            return meta
+        if proc_aman is None:
+            proc_aman = meta.preprocess
+        keep = ~has_all_cut(proc_aman.fp_flags.fp_flags)
+        meta.restrict("dets", meta.dets.vals[keep])
+        return meta
+
+class BadSubscanFlags(_Preprocess):
+    """Identifies and flags bad subscans.
+
+      Example config block::
+
+        - name : "bad_subscan_flags"
+          calc: 
+            merge: True
+            overwrite: True
+          save: True
+    
+    .. autofunction:: sotodlib.tod_ops.flags.get_badsubscan_flags
+    """
+    name = "bad_subscan_flags"
+    
+    def calc_and_save(self, aman, proc_aman):
+        msk_ss, msk_det = tod_ops.flags.get_badsubscan_flags(aman, **self.calc_cfgs)
+        ss_aman = core.AxisManager(aman.dets, aman.samps)
+        ss_aman.wrap("bad_subscan_flags", msk_ss, [(0, 'dets'), (1, 'samps')])
+        self.save(proc_aman, ss_aman)
+    
+    def save(self, proc_aman, ss_aman): #, ssdet_aman):
+        if self.save_cfgs is None:
+            return
+        if self.save_cfgs:
+            proc_aman.wrap("badsubscan_flags", ss_aman)
+
+class NoiseFlags(_Preprocess):
```

Comment:
Right, I guess in `processes.Noise` the selection
```
keep = wn <= np.float64(self.select_cfgs["max_noise"])
        meta.restrict("dets", meta.dets.vals[keep])
```
is already equivalent to the white noise cuts I do here. Do we want to add the fknee cuts as well (which is not strictly critical) or just keep the white noise ones (which seem to be critical to avoid flat maps)?